torchbenchmark/e2e_models/hf_bert/__init__.py

```diff
@@ -56,6 +57,21 @@ def __init__(self, test, batch_size=None, extra_args=[]):
                   "--num_train_epochs", num_train_epochs,
                   "--output_dir", output_dir]
         hf_args = parse_args(in_arg)
+
+        # ideally we don't modify the model code directly, but attaching deepspeed
```

Comment:
This should be part of `apply_deepspeed()`

Reply:
well.. i thought about this at first, but the actual deepspeed config is kind of model-specific.  I am not sure how to do it, but, for instance both the stage, bucket_size and possibly other settings should be customized based on the model size / #params, etc.

i'm not sure how to write a generic 'apply_deepspeed' function.

On the input side, the function would need the model itself to provide the deepspeed config for that model, so we could have `apply_deepspeed(model_deepspeed_config_dict)`.

but then also, we have to hook the deepspeed up to the model in a model-specific way.  For huggingface, we have to set the DeepSpeed plugin inside the Accelerator ctor.

How would you propose to do this inside apply_deepspeed?

I guess one option is that apply_deepspeed has different logic depending on model type, and hardcodes the model-specific settings as well as the logic for what to modify in the model?

There is still an issue of sequencing;  in the hf case, we have to configure deepspeed part-way-through the model __init__ so it is before the Accelerator() ctor in prepare.  Unless we refactor that a bit, it'd be hard to call `apply_deepspeed` outside of the model __init__.  (Ideally, what I woudl have liked to do is in util/deepspeed.py, just build the model, then 'apply deepspeed' to it.  But this seemed impossible given how hf_ model is architected.)